### PR TITLE
[BugFix] fix crash of json array compare

### LINKED ISSUE #9316: BE crash when compare JSON array
### Steps to reproduce the behavior (Required)

Try to compare JSON array but with different length would make the BE crash:

```sql
select * from tj1 where j1 != '[]' limit 1;
```

### StarRocks version (Required)
- Since 2.2

### PR DESCRIPTION
**Pull request by murphyatwork**
## What type of PR is this：
- [x] bug
- [ ] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #9316

### CODE REVIEW

**Comment by mofeiatwork** [comment]
@Mergifyio backport branch-2.2 branch-2.3

File: be/src/util/json.cpp [hunk 1]
```diff
@@ -168,20 +168,18 @@ static int sliceCompare(const vpack::Slice& left, const vpack::Slice& right) {
...
+            auto left_item = left.at(i);
+            auto right_item = right.at(i);
+            int x = sliceCompare(left_item, right_item);
+            if (x != 0) {
+                return x;
             }
-            idx++;
         }
-        return 0;
+        return left.length() - right.length();
```
Comment (stdpain) [inline be/src/util/json.cpp:182]: maybe we can compare at first. then we can reduce some unused compare costs
```
return l.length() == r.length() && l.children().equals(r.children());
```

Comment (mofeiatwork) [reply be/src/util/json.cpp:182]: fixed.

**Review by kangkaisen — APPROVED** [review]

**Review by stdpain — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by stdpain** [comment]
run starrocks_admit_test

**Comment by mofeiatwork** [comment]
run starrocks_admit_test

**Pull request merged**